freqtrade/freqtradebot.py

```diff
@@ -904,18 +904,18 @@ def execute_entry(
 
     def cancel_stoploss_on_exchange(self, trade: Trade) -> Trade:
         # First cancelling stoploss on exchange ...
-        if trade.stoploss_order_id:
-            try:
-                logger.info(f"Canceling stoploss on exchange for {trade}")
-                co = self.exchange.cancel_stoploss_order_with_result(
-                    trade.stoploss_order_id, trade.pair, trade.amount)
-                self.update_trade_state(trade, trade.stoploss_order_id, co, stoploss_order=True)
-
-                # Reset stoploss order id.
-                trade.stoploss_order_id = None
-            except InvalidOrderException:
-                logger.exception(f"Could not cancel stoploss order {trade.stoploss_order_id} "
-                                 f"for pair {trade.pair}")
+        if trade.has_open_sl_orders:
+            for o in trade.orders:
```

Comment:
you will want to use a helper here i'd think (similar to the `.open_orders` one, but for stoploss orders).
Otherwise you may end up distributing this same logic (loop all, skip most) throughout the bot,

Reply:
Yes, I hesitated but I pushed like that to get your opinion 👍🏽